tests/unit_tests/test_solver.cpp

```diff
@@ -1,135 +1,172 @@
 #include <limits>
 
-#include <gtest/gtest.h>
 #include <Kokkos_Core.hpp>
+#include <gtest/gtest.h>
 
 #include "src/rigid_pendulum_poc/solver.h"
 
 namespace openturbine::rigid_pendulum::tests {
 
-Kokkos::View<double**, Kokkos::DefaultHostExecutionSpace> 
-create_diagonal_matrix(std::vector<double> values)
-{
-  auto matrix = Kokkos::View<double**, Kokkos::DefaultHostExecutionSpace>("matrix", values.size(), values.size());
-  auto diagonal_entries = Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, values.size());
-  auto fill_diagonal = [matrix, values](int index) { matrix(index, index) = values[index]; };
+Kokkos::View<double**, Kokkos::DefaultHostExecutionSpace> create_diagonal_matrix(
+    const std::vector<double>& values) {
+    auto matrix = Kokkos::View<double**, Kokkos::DefaultHostExecutionSpace>("matrix", values.size(),
+                                                                            values.size());
+    auto diagonal_entries =
+        Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, values.size());
+    auto fill_diagonal = [matrix, values](int index) {
+        matrix(index, index) = values[index];
+    };
 
-  Kokkos::deep_copy(matrix, 0);
-  Kokkos::parallel_for(diagonal_entries, fill_diagonal);
+    Kokkos::deep_copy(matrix, 0);
+    Kokkos::parallel_for(diagonal_entries, fill_diagonal);
 
-  return matrix;
+    return matrix;
 }
 
-Kokkos::View<double*, Kokkos::DefaultHostExecutionSpace>
-create_vector(std::vector<double> values)
-{
-  auto vector = Kokkos::View<double*, Kokkos::DefaultHostExecutionSpace>("vector", values.size());
-  auto entries = Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, values.size());
-  auto fill_vector = [vector, values](int index) { vector(index) = values[index]; };
+Kokkos::View<double*, Kokkos::DefaultHostExecutionSpace> create_vector(
+    const std::vector<double>& values) {
```

Comment:
This formatting is not great.  We should change our clang-format to make it better (I think putting the line break after the type instead of in the argument list is more readable).  Most of the format changes are just stylistic differences, but ones like this risk making our production code harder to read, in my opinion

Reply:
Fair enough. I'll play around with it and figure something out in parallel. We can probably move forward with this PR.